docs/adr/0008_signing_verification.md

```diff
@@ -0,0 +1,623 @@
+         normalisationAlgorithm: jsonNormalisation/v1
+         value: cf08abae08bb874597630bc0573d941b1becc92b4916cbe3bef9aa0e89aec3f6
+       signature:
+         algorithm: RSASSA-PKCS1-V1_5
+         mediaType: application/vnd.ocm.signature.rsa
+         value: 390157b7...75ab2705d6
+   ```
+3. Verify the signature using the configured verifier from `.ocmconfig`.
+
+### Signing
+
+Signing uses the analogous command:
+
+```shell
+ocm sign componentversion --signature mysig --signer rsapss ghcr.io/open-component-model/ocm//ocm.software/ocm:0.17.0
+```
+
+This downloads, normalizes, signs, and re-uploads the descriptor. Signing and verification configs live in `.ocmconfig`.
+
+#### Two-Stage Signing
+
+In most cases, normalizing/digesting and signing can be done in separate steps.
+This is useful for cases where the descriptor is generated and hashed in a CI, which has a separate step for building
+and signing.
+
+In these cases, users can choose to add a digest to the descriptor and sign it separately:
+
+```shell
+ocm add digest {--signature default} {--hash-algorithm=sha256} {--normalisation-agorithm=jsonNormalisation/v1} {--upload=true} ghcr.io/open-component-model/ocm//ocm.software/ocm:0.17.0
+
+> name: default
+> digest:
+>   hashAlgorithm: sha256
+>   normalisationAlgorithm: jsonNormalisation/v1
+>   value: cf08abae08bb874597630bc0573d941b1becc92b4916cbe3bef9aa0e89aec3f6
+```
+
+After this, as long as `--upload` is set, the descriptor will be updated with this signature field, containing a digest,
+but no actual signature value.
+
+This can then be used to sign the descriptor against a pinned signature:
+
+```shell
+ocm sign componentversion ghcr.io/open-component-model/ocm//ocm.software/ocm:0.17.0 --signature default@cf08abae08bb874597630bc0573d941b1becc92b4916cbe3bef9aa0e89aec3f6 --signer rsapss
+```
+
+This will cause the descriptor to be downloaded, and the digest will be searched by the matching signature hash
+algorithm and value.
+Because the descriptor is not yet signed, the signature will be submitted to the given signer and the descriptor will be
+updated with the signed digest afterwards, which will make the descriptor ready for verification. (The normalisation and
+hash algorithm are derived from the digest specification)
+
+```mermaid
+sequenceDiagram
+    participant U as User
+    participant CLI as ocm CLI
+    participant Repo as OCM Repository
+    participant Handler as Signer Handler
+    participant Creds as Credentials Store
+
+    Note over U,CLI: Step 1: Add Digest
+    U->>CLI: ocm add digest --signature default --upload
+    CLI->>Repo: Download Component Descriptor
+    CLI->>CLI: Normalize + Compute Digest
+    CLI->>Repo: Upload Descriptor with Digest only (no signature)
+    CLI-->>U: Digest field added to descriptor
+
+    Note over U,CLI: Step 2: Sign Pinned Digest
+    U->>CLI: ocm sign componentversion --signature default@<digest> --signer rsapss
+    CLI->>Repo: Download Descriptor (with digest field)
```

Comment:
If download is a request to the `Repo` shouldn't there be a response from the Repo here depicting the returned unsigned descriptor?

Reply:
you are right as always! Will add response ways